src/pages/MyPatches.tsx

```diff
@@ -1,5 +1,30 @@
 import React from "react";
+import { useLocation, useHistory } from "react-router-dom";
+import queryString from "query-string";
+import Checkbox from "@leafygreen-ui/checkbox";
 
 export const MyPatches = () => {
-  return <div>My Patches Page</div>;
+  const { search, pathname } = useLocation();
+  const history = useHistory();
+  const parsed = queryString.parse(search);
+  let showCommitQueue = parsed.commitQueue === "true";
+
+  function onCheckboxChange() {
+    showCommitQueue = !showCommitQueue;
+    history.push({
+      pathname: pathname,
+      search: "?commitQueue=" + showCommitQueue
```

Comment:
this line will have to be updated as we add more query params to the page which could make things confusing to maintain later on. Check out this function that passes on existing query params https://github.com/evergreen-ci/spruce/blob/new-ui-feature-branch/src/pages/patch/patchTabs/tasks/TasksTable.tsx#L31